ogr/ogrsf_frmts/miramon/ogrmiramonlayer.cpp

```diff
@@ -0,0 +1,2613 @@
+            phMiraMonLayer->bIsReal3d = 1;
+        }
+
+        hMMFeature.nICoord++;
+    }
+    hMMFeature.nIRing++;
+    hMMFeature.nNRings++;
+    return OGRERR_NONE;
+}
+
+/****************************************************************************/
+/*                           MMLoadGeometry()                               */
+/*                                                                          */
+/*      Loads on a MiraMon object Feature all coordinates from feature      */
+/*                                                                          */
+/****************************************************************************/
+OGRErr OGRMiraMonLayer::MMLoadGeometry(OGRGeometryH hGeom)
+
+{
+    OGRErr eErr = OGRERR_NONE;
+    MM_BOOLEAN bExternalRing;
+
+    /* -------------------------------------------------------------------- */
+    /*      This is a geometry with sub-geometries.                         */
+    /* -------------------------------------------------------------------- */
+    int nGeom = OGR_G_GetGeometryCount(hGeom);
+
+    int eLT = LOG_ACTION(wkbFlatten(OGR_G_GetGeometryType(hGeom)));
+
+    if (eLT == wkbMultiPolygon || eLT == wkbPolyhedralSurface ||
+        eLT == wkbTIN || eLT == wkbTriangle)
+    {
+        for (int iGeom = 0; iGeom < nGeom; iGeom++)
+        {
+            OGRGeometryH poNewGeometry = OGR_G_GetGeometryRef(hGeom, iGeom);
+
+            // Reads all coordinates
+            eErr = MMLoadGeometry(poNewGeometry);
+            if (eErr != OGRERR_NONE)
+                return eErr;
+        }
+    }
+    else if (eLT == wkbPolygon)
+    {
+        for (int iGeom = 0; iGeom < nGeom && eErr == OGRERR_NONE; iGeom++)
+        {
+            OGRGeometryH poNewGeometry = OGR_G_GetGeometryRef(hGeom, iGeom);
+
+            if (iGeom == 0)
+                bExternalRing = true;
+            else
+                bExternalRing = false;
+
+            eErr = MMDumpVertices(poNewGeometry, bExternalRing, TRUE);
+            if (eErr != OGRERR_NONE)
+                return eErr;
+        }
+    }
+    else if (eLT == wkbPoint || eLT == wkbLineString)
+    {
+        // Reads all coordinates
+        eErr = MMDumpVertices(hGeom, true, FALSE);
+
+        if (eErr != OGRERR_NONE)
+            return eErr;
+    }
+    else if (eLT == wkbGeometryCollection)
+    {
+        CPLError(
+            CE_Failure, CPLE_NotSupported,
+            "MiraMon: wkbGeometryCollection inside a wkbGeometryCollection?");
+        return OGRERR_UNSUPPORTED_GEOMETRY_TYPE;
+    }
+
+    return OGRERR_NONE;
+}
+
+/****************************************************************************/
+/*                           WriteGeometry()                                */
+/*                                                                          */
+/*      Write a geometry to the file.  If bExternalRing is true it          */
+/*      means the ring is being processed is external.                      */
```

Comment:
the sentence about bExternalRing must be a copy&paste left over

Reply:
Sure. Thanks